packages/flutter_tools/lib/src/device.dart

```diff
@@ -175,11 +175,27 @@ class DeviceManager {
   Future<List<Device>> findTargetDevices(FlutterProject flutterProject) async {
     List<Device> devices = await getDevices().toList();
 
-    if (devices.length > 1 && !deviceManager.hasSpecifiedAllDevices && !deviceManager.hasSpecifiedDeviceId) {
-      devices = devices
-          .where((Device device) => isDeviceSupportedForProject(device, flutterProject))
-          .toList();
+    // Always remove web and fuchsia devices from `--all`.
+    if (hasSpecifiedAllDevices) {
+      devices = <Device>[
+        for (Device device in devices)
+          if (await device.targetPlatform != TargetPlatform.fuchsia
+          && await device.targetPlatform != TargetPlatform.web_javascript)
+            device
+      ];
+    }
+
+    // Remove unsupported devices for project.
+    if (devices.length > 1 && !hasSpecifiedDeviceId) {
+      devices = <Device>[
+        for (Device device in devices)
+          if (device.isSupportedForProject(flutterProject))
```

Comment:
why not check if the device is supported in `device.isSupportedForProject` ?

Reply:
I'm not sure what you mean - we do check if the device is supported in `device.isSupportedForProject` ?

Reply:
I was wondering why not using `device.isSupportedForProject` to skip `web` and `fuchsia`.

Reply:
We only remove `web` and `fuchsia` when `hasSpecifiedAllDevices` is true. This is because the all devices flag will attempt to share a frontend_server instance, which may fail with these devices since they require differently configured compilers than mobile or desktop.

Reply:
Thanks for the clarification. maybe that can be in comment? 🙂